x-pack/elastic-agent/magefile.go

```diff
@@ -579,6 +579,10 @@ func packageAgent(requiredPackages []string, packagingFn func()) {
 		}
 
 		os.Setenv(agentDropPath, dropPath)
+		if runtime.GOARCH == "arm64" {
+			os.Setenv("PLATFORMS", runtime.GOOS+"/"+runtime.GOARCH)
+			defer os.Unsetenv("PLATFORMS")
+		}
```

Comment:
Maybe we don't want to unset the variable if it was set.

e.g.

```
const platformsVar = "PLATFORMS"
oldPlatforms := os.GetEnv(platformsVar)
os.Setenv(...)
if oldPlatforms != "" {
  defer os.Setenv(platformsVar, oldPlatform)
} else {
  defer os.Unsetenv(platofrmsVar)
}
```

Reply:
in arm64 is not defined ever, in any case, I will add the backup logic